File: backend/src/main/java/com/shoptourr/api/v1/dto/taxfree/TaxFreeDtos.java
package com.shoptourr.api.v1.dto.taxfree;

import com.shoptourr.api.v1.dto.common.CommonDtos.MoneyDto;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

/**
 * Tax Free screen — eligibility against country thresholds (mock TAXFREE_INFO).
 */
public final class TaxFreeDtos {

    private TaxFreeDtos() {}

    public record TaxFreeRulesDto(
            String currency,
            MoneyDto minimumPurchase,
            /** Fraction e.g. 0.13 = 13% estimated refund. */
            BigDecimal estimatedRefundRate,
            String regionLabel
    ) {}

    public record TaxFreeEligibleItemDto(
            UUID purchaseId,
            String name,
            MoneyDto amount,
            MoneyDto estimatedRefund,
            boolean meetsMinimum
    ) {}

    public record TaxFreeSummaryDto(
            UUID tripId,
            TaxFreeRulesDto rules,
            int eligibleCount,
            MoneyDto eligibleTotal,
            MoneyDto estimatedRefundTotal,
            /** How much more spend needed to unlock next eligible item / form, if any. */
            MoneyDto remainingToMinimum,
            List<TaxFreeEligibleItemDto> items
    ) {}
}

File: shared/src/commonMain/kotlin/com/example/shoptourr/api/taxfree/TaxFreeDtos.kt
package com.example.shoptourr.api.taxfree

import com.example.shoptourr.api.common.MoneyDto

data class TaxFreeRulesDto(
    val currency: String,
    val minimumPurchase: MoneyDto,
    val estimatedRefundRate: String,
    val regionLabel: String,
)

data class TaxFreeEligibleItemDto(
    val purchaseId: String,
    val name: String,
    val amount: MoneyDto,
    val estimatedRefund: MoneyDto,
    val meetsMinimum: Boolean,
)

data class TaxFreeSummaryDto(
    val tripId: String,
    val rules: TaxFreeRulesDto,
    val eligibleCount: Int,
    val eligibleTotal: MoneyDto,
    val estimatedRefundTotal: MoneyDto,
    val remainingToMinimum: MoneyDto? = null,
    val items: List<TaxFreeEligibleItemDto>,
)
